### PR TITLE
Add required terms checkbox to newsletter signup

### PR DESCRIPTION
**Pull request by marineb**

### CODE REVIEW

File: src/Form/NewsletterSubscriptionType.php [hunk 1]
```diff
@@ -24,6 +25,9 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'required' => false,
                 'filter_emojis' => true,
             ])
+            ->add('conditions', CheckboxType::class, [
+                'required' => false,
```
Comment (HeahDude) [inline src/Form/NewsletterSubscriptionType.php]: Could you please add a `constraints` option like:
```php
->add('conditions', CheckboxType::class, [
    'required' => false,
    'constraints' => new IsTrue(['message' => 'newsletter.conditions.required']),
    'mapped' => false,
])
```
and remove the public property on the object?

I can handle it if you want.

File: app/Resources/translations/validators.fr.yml [hunk 2]
```diff
@@ -88,12 +88,14 @@ donation.amount.less_than_7500: La valeur maximale d'un don est de 7500 €.
 #
```
Comment (HeahDude) [inline app/Resources/translations/validators.fr.yml]: The file has moved to `/translations/validators.fr.yml`

File: app/Resources/views/newsletter/subscription.html.twig [hunk 3]
```diff
@@ -12,15 +12,12 @@
         <section>
```
Comment (HeahDude) [inline app/Resources/views/newsletter/subscription.html.twig]: This one to `/templates/newsletter/subscription.html.twig`

**Comment by marineb** [comment]
@HeahDude I didn't know what you meant by removing the public property on the object. Would love to learn know! :)

File: src/Entity/NewsletterSubscription.php [hunk 4]
```diff
@@ -51,12 +51,17 @@ class NewsletterSubscription
...
+     *     minMessage="newsletter.postalCode.invalid",
+     *     maxMessage="newsletter.postalCode.invalid"
      * )
      */
     private $postalCode;
 
+    /**
+     * @Assert\IsTrue(message="newsletter.conditions.required")
+     */
+    public $conditions;
```
Comment (HeahDude) [inline src/Entity/NewsletterSubscription.php]: Now that there is a non mapped field in the form this property can be removed.

File: src/Form/NewsletterSubscriptionType.php [hunk 5]
```diff
@@ -24,6 +25,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                'constraints' => new IsTrue(['message' => 'newsletter.conditions.required']),
```
Comment (HeahDude) [inline src/Form/NewsletterSubscriptionType.php]: The CI says that this `IsTrue` class is missing its use statement.

**Pull request closed without merging**